backend/lib/spark-api-submit-deals.js

```diff
@@ -45,7 +49,7 @@ export const findAndSubmitUnsubmittedDeals = async (pgPool, batchSize, submitDea
  *
  * @param {PgPool} pgPool
  * @param {number} batchSize
- * @returns {AsyncGenerator<Array>}
+ * @returns {AsyncGenerator<Array<Static<typeof SubmittableDeal>>, void, unknown>}
```

Comment:
```suggestion
 * @returns {AsyncGenerator<SubmittableDeal[], void, unknown>}
```

I think same [comment ](https://github.com/CheckerNetwork/spark-deal-observer/pull/95/files#r1952235257) is applicable here.